src/Simplex/Messaging/Agent/Protocol.hs

```diff
@@ -473,18 +477,20 @@ instance ToJSON QueueDirection where
 instance FromJSON QueueDirection where
   parseJSON = strParseJSON "QueueDirection"
 
-data SwitchPhase = SPStarted | SPConfirmed | SPCompleted
+data SwitchPhase = SPStarted | SPConfirmed | SPFinalizing | SPCompleted
```

Comment:
SPFinalizing - not clear what it actually means... Need to check when exactly it happens.